File: sdk-ts/src/adapters/compat.ts
/**
 * Shared helper for the openai/anthropic adapters: both SDKs' tool-call
 * objects can appear either as plain objects (raw API JSON) or as SDK
 * instances, depending on version and call site. This reads either shape
 * without requiring either SDK as a dependency.
 */
export function getField<T = unknown>(obj: unknown, key: string): T | undefined {
  if (obj && typeof obj === "object") {
    return (obj as Record<string, T>)[key];
  }
  return undefined;
}

File: sdk-python/agentguard/adapters/_compat.py
"""Shared helper for the openai/anthropic adapters: both SDKs expose tool
call objects either as dicts (raw API JSON) or as attribute-bearing SDK
objects, depending on version and call site. This reads either shape without
requiring either SDK as a dependency.
"""
from __future__ import annotations

from typing import Any, Optional


def get_field(obj: Any, key: str) -> Optional[Any]:
    if isinstance(obj, dict):
        return obj.get(key)
    return getattr(obj, key, None)
